packages/babel/test/as-output-plugin.js

```diff
@@ -0,0 +1,345 @@
+import * as nodePath from 'path';
+
+import test from 'ava';
+import { rollup } from 'rollup';
+import { SourceMapConsumer } from 'source-map';
+
+import babelPlugin from '..';
+
+process.chdir(__dirname);
+
+function getLocation(source, charIndex) {
+  const lines = source.split('\n');
+  const len = lines.length;
+
+  let lineStart = 0;
+  let i;
+
+  for (i = 0; i < len; i += 1) {
+    const line = lines[i];
+    // +1 for newline
+    const lineEnd = lineStart + line.length + 1;
+
+    if (lineEnd > charIndex) {
+      return { line: i + 1, column: charIndex - lineStart };
+    }
+
+    lineStart = lineEnd;
+  }
+
+  throw new Error('Could not determine location of character');
+}
+
+function replaceConsoleLogProperty({ types: t }) {
+  return {
+    name: 'replace-console-log-property',
+    visitor: {
+      MemberExpression(path, state) {
+        const { opts } = state;
+        if (path.node.property.name === 'log') {
+          path.get('property').replaceWith(t.identifier(opts.replace));
+        }
+      }
+    }
+  };
+}
+
+async function generate(input, babelOptions = {}, generateOptions = {}, rollupOptions = {}) {
```

Comment:
In cases where only `code` is used from `generated`, `getCode` could be used instead.